### PR TITLE
Cover additional scenarios for product show page and add missing data

### PR DESCRIPTION
**Pull request by TheMilek**
| Q               | A
|-----------------|-----
| Branch?         | 2.0 
| Bug fix?        | a bit
| New feature?    | no
| BC breaks?      | no
| License         | MIT

Some missing information has been added
Screenshot 2024-07-10 at 17 40 03
Screenshot 2024-07-10 at 17 45 13

### CODE REVIEW

File: features/admin/product/viewing_product_in_admin_panel/accessing_product_edit_page_from_show_page.feature [hunk 1]
```diff
@@ -12,13 +12,13 @@ Feature: Accessing the product edit page from the show page
         And I am logged in as an administrator
         And I am browsing products
 
-    @todo @ui @no-api
+    @ui @no-api
```
Comment (GSadee) [inline features/admin/product/viewing_product_in_admin_panel/accessing_product_edit_page_from_show_page.feature]: Change the order of tags here and in other scenarios
```suggestion
    @no-api @ui
```

File: src/Sylius/Behat/Element/Product/ShowPage/AttributesElement.php [hunk 2]
```diff
@@ -17,27 +17,30 @@
...
 
-    public function hasNonTranslatableAttribute(string $attribute, string $value): bool
+    public function hasNonTranslatableAttribute(string $attribute, float|string $value): bool
     {
-        $attributeValue = $this->getDocument()->find('css', sprintf('.ui.segment[data-tab="non-translatable"] tr:contains("%s") td:nth-child(2)', $attribute))->getText();
+        $hasName = $this->hasElement('non_translatable_attribute_name', ['%name%' => $attribute]);
+        $hasValue = $this->hasElement('non_translatable_attribute_value', ['%value%' => $value]);
 
-        return str_contains($attributeValue, $value);
+        return $hasName && $hasValue;
```
Comment (GSadee) [inline src/Sylius/Behat/Element/Product/ShowPage/AttributesElement.php:33]: Are we checking the same attribute in this way? 🤔

File: src/Sylius/Behat/Element/Product/ShowPage/DetailsElement.php [hunk 3]
```diff
@@ -61,10 +55,10 @@ public function getProductTaxCategory(): string
     protected function getDefinedElements(): array
     {
         return array_merge(parent::getDefinedElements(), [
-            'channels' => '#details tr:contains("Channels") td:nth-child(2)',
-            'current_stock' => '#details tr:contains("Current stock") td:nth-child(2)',
-            'product_code' => '#details tr:contains("Code") td:nth-child(2)',
-            'tax_category' => '#details tr:contains("Tax category") td:nth-child(2)',
+            'channel' => '[data-test-channel="%channelCode%"]',
```
Comment (GSadee) [inline src/Sylius/Behat/Element/Product/ShowPage/DetailsElement.php]:
```suggestion
            'channel' => '[data-test-channel="%channel_code%"]',
```

File: src/Sylius/Behat/Element/Product/ShowPage/AttributesElement.php [hunk 4]
```diff
@@ -17,27 +17,30 @@
...
     }
 
     protected function getDefinedElements(): array
     {
         return array_merge(parent::getDefinedElements(), [
-            'attribute_value_in_locale' => '.attributes:contains("%locale%") ~ table tr:contains("%attribute%") td:nth-child(2)',
-            'attributes_in_locale' => '#attributes :contains("%locale%")',
+            'attribute_name' => '[data-test-attribute-name="%name%"]',
+            'attribute_value' => '[data-test-attribute-value]',
+            'attribute_with_locale' => '[data-test-attribute-with-locale="%locale%"]',
```
Comment (GSadee) [inline src/Sylius/Behat/Element/Product/ShowPage/AttributesElement.php]:
```suggestion
            'attribute_with_locale' => '[data-test-attribute-with-locale="%locale_code%"]',
```

File: src/Sylius/Behat/Element/Product/ShowPage/PricingElement.php [hunk 5]
```diff
@@ -86,19 +87,24 @@ protected function getDefinedElements(): array
         return array_merge(parent::getDefinedElements(), [
             'simple_product_pricing_row' => '[data-test-simple-product="%channelCode%"]',
             'variant_pricing_row' => '[data-test-variant-pricing="%channelCode%.%variantCode%"]',
+            'price_row' => '[data-test-pricing="%channelCode%"]',
```
Comment (GSadee) [inline src/Sylius/Behat/Element/Product/ShowPage/PricingElement.php]: Not alphabetical order 😅

Comment (GSadee) [inline src/Sylius/Behat/Element/Product/ShowPage/PricingElement.php]:
```suggestion
            'price_row' => '[data-test-pricing="%channel_code%"]',
```

File: src/Sylius/Behat/Page/Admin/Product/ShowPage.php [hunk 6]
```diff
@@ -81,12 +81,12 @@ public function showVariantEditPage(ProductVariantInterface $variant): void
     protected function getDefinedElements(): array
     {
         return array_merge(parent::getDefinedElements(), [
-            'breadcrumb' => '.breadcrumb > div',
-            'edit_product_button' => '#edit-product',
-            'edit_variant_button' => '#variants .variants-accordion__title:contains("%variantCode%") .edit-variant',
-            'product_name' => '#header h1 .content > span',
+            'breadcrumb' => '.breadcrumb',
+            'edit_product_button' => '[data-test-edit-product-button]',
+            'edit_variant_button' => '[data-test-edit-variant-button="%variantCode%"]',
```
Comment (GSadee) [inline src/Sylius/Behat/Page/Admin/Product/ShowPage.php]:
```suggestion
            'edit_variant_button' => '[data-test-edit-variant-button="%variant_code%"]',
```

File: src/Sylius/Bundle/AdminBundle/templates/product/common/header/title_block/actions/edit_product.html.twig [hunk 7]
```diff
@@ -0,0 +1,10 @@
+{% from '@SyliusAdmin/shared/helper/icon.html.twig' import icon as _icon %}
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/templates/product/common/header/title_block/actions/edit_product.html.twig]:
```suggestion
{% from '@SyliusAdmin/shared/helper/icon.html.twig' import icon %}
```

File: src/Sylius/Bundle/AdminBundle/templates/product/show/content/page_body/attributes.html.twig [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+<div class="card mb-3">
+    <div class="card-header">
+        <div class="card-title">
+            {{ 'sylius.ui.attributes'|trans }}
+        </div>
+    </div>
+    <div class="card-body" {{ sylius_test_html_attribute('attributes') }}>
+        <div class="bg-white mb-3">
+            <div class="accordion" id="attribute-translations">
+                {% set setLocales = [] %}
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/templates/product/show/content/page_body/attributes.html.twig]:
```suggestion
                {% set set_locales_codes = [] %}
```
or
```suggestion
                {% set used_locales_codes = [] %}
```

File: src/Sylius/Bundle/AdminBundle/templates/product/show/content/page_body/attributes.html.twig [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+                {% for attribute in product.attributes %}
+                    {% if attribute.localeCode not in setLocales %}
+                        {% set locale = attribute.localeCode %}
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/templates/product/show/content/page_body/attributes.html.twig]:
```suggestion
                        {% set locale_code = attribute.localeCode %}
```

File: src/Sylius/Bundle/AdminBundle/templates/product/show/content/page_body/attributes.html.twig [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+                            <div class="accordion-item">
+                                <h2 class="accordion-header">
+                                    <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#translation-{% if attribute.localeCode %}{{ locale }}{% else %}non-translatable{% endif%}" aria-expanded="{% if loop.first %}true{% else %}false{% endif %}" aria-controls="translation-{% if attribute.localeCode %}{{ locale }}{% else %}non-translatable{% endif %}">
+                                        {% if attribute.localeCode %}
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/templates/product/show/content/page_body/attributes.html.twig]: You have set the variable `locale`

File: src/Sylius/Bundle/AdminBundle/templates/product/show/content/page_body/attributes.html.twig [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+    <div class="card-body" {{ sylius_test_html_attribute('attributes') }}>
+        <div class="bg-white mb-3">
+            <div class="accordion" id="attribute-translations">
+                {% set setLocales = [] %}
+                {% for attribute in product.attributes %}
+                    {% if attribute.localeCode not in setLocales %}
+                        {% set locale = attribute.localeCode %}
+                            <div class="accordion-item">
+                                <h2 class="accordion-header">
+                                    <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#translation-{% if attribute.localeCode %}{{ locale }}{% else %}non-translatable{% endif%}" aria-expanded="{% if loop.first %}true{% else %}false{% endif %}" aria-controls="translation-{% if attribute.localeCode %}{{ locale }}{% else %}non-translatable{% endif %}">
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/templates/product/show/content/page_body/attributes.html.twig]: As there are a lot of code checks, maybe divide this code into the one for the translatable attribute and the non-translatable one

File: src/Sylius/Bundle/AdminBundle/templates/product/show/content/page_body/attributes/name.html.twig [hunk 12]
```diff
@@ -0,0 +1,13 @@
+{% set attributes = hookable_metadata.context.attributes  %}
+
+<div class="col-12 col-md-4">
+    {% if hookable_metadata.context.locale is defined %}
+        {% for attributeValue in attributes|filter(attributeValue => attributeValue.localeCode == hookable_metadata.context.locale) %}
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/templates/product/show/content/page_body/attributes/name.html.twig]:
```suggestion
        {% for attribute_value in attributes|filter(attribute_value => attribute_value.localeCode == hookable_metadata.context.locale) %}
```

File: src/Sylius/Bundle/AdminBundle/templates/product/show/content/page_body/attributes/name.html.twig [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+            <p {{ sylius_test_html_attribute('attribute-name', "%s"|format(attributeValue.name)) }}>{{ attributeValue.name }}</p>
+        {% endfor %}
+    {% else %}
+        {% for attributeValue in attributes|filter(attributeValue => attributeValue.localeCode == null) %}
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/templates/product/show/content/page_body/attributes/name.html.twig]:
```suggestion
        {% for attribute_value in attributes|filter(attribute_value => attribute_value.localeCode == null) %}
```

File: src/Sylius/Bundle/AdminBundle/templates/product/show/content/page_body/attributes/value.html.twig [hunk 14]
```diff
@@ -0,0 +1,30 @@
+{% set attributes = hookable_metadata.context.attributes  %}
+
+<div class="col-12 col-md-8">
+    {% if hookable_metadata.context.locale is defined %}
+        {% for attributeValue in attributes|filter(attributeValue => attributeValue.localeCode == hookable_metadata.context.locale) %}
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/templates/product/show/content/page_body/attributes/value.html.twig]: Use snake_case here and below

File: features/admin/product/viewing_product_in_admin_panel/viewing_details_of_product_with_variants.feature [hunk 15]
```diff
@@ -32,32 +32,32 @@ Feature: Viewing details of a product with variants
...
-    @todo @ui @no-api
+    @no-api @ui
     Scenario: Viewing variants' details
         When I access the "Iron Shield" product
         Then I should see 2 variants
         And I should see "Iron Shield - very big" variant with code "123456789-xl", priced "$25.00" and current stock 5 and in "United States" channel
         And I should see "Iron Shield - very small" variant with code "123456789-xs", priced "$15.00" and current stock 12 and in "United States" channel
 
-    @todo @ui @javascript @api
+    @api @mink:chromedriver @ui
```
Comment (Wojdylak) [inline features/admin/product/viewing_product_in_admin_panel/viewing_details_of_product_with_variants.feature]:
```suggestion
    @api @ui @mink:chromedriver
```

File: features/admin/product/viewing_product_in_admin_panel/viewing_details_of_simple_product.feature [hunk 16]
```diff
@@ -65,13 +65,13 @@ Feature: Viewing details of a simple product
         And I should see product's depth is 20
         And I should see product's weight is 25
 
-    @todo @ui @javascript @api
+    @api @mink:chromedriver @ui
```
Comment (Wojdylak) [inline features/admin/product/viewing_product_in_admin_panel/viewing_details_of_simple_product.feature]:
```suggestion
    @api @ui @mink:chromedriver
```

File: src/Sylius/Behat/Element/Product/ShowPage/AttributesElement.php [hunk 17]
```diff
@@ -17,27 +17,32 @@
 
 final class AttributesElement extends Element implements AttributesElementInterface
 {
-    public function hasAttributeInLocale(string $attribute, string $locale, string $value): bool
+    public function hasAttributeInLocale(string $attribute, string $localeCode, string $value): bool
     {
-        $values = $this->getDocument()->find('css', sprintf('#attributes .tab.segment[data-tab="%s"]', $locale));
+        $attributeValue = $this->getDocument()->find('css', sprintf('[data-test-attribute-with-locale="%s"]', $localeCode));
```
Comment (Wojdylak) [inline src/Sylius/Behat/Element/Product/ShowPage/AttributesElement.php]:
```suggestion
        $attributeValue = $this->getElement('attribute_with_locale', ...);
```

File: src/Sylius/Behat/Element/Product/ShowPage/AttributesElement.php [hunk 18]
```diff
@@ -17,27 +17,32 @@
...
         return array_merge(parent::getDefinedElements(), [
-            'attribute_value_in_locale' => '.attributes:contains("%locale%") ~ table tr:contains("%attribute%") td:nth-child(2)',
-            'attributes_in_locale' => '#attributes :contains("%locale%")',
+            'attribute_name' => '[data-test-attribute-name="%name%"]',
+            'attribute_value' => '[data-test-attribute-value]',
+            'attribute_with_locale' => '[data-test-attribute-with-locale="%locale_code%"]',
+            'attribute_without_locale' => '[data-test-attribute-without-locale]',
+            'non_translatable-attribute' => '[data-test-non-translatable-attribute]',
+            'non_translatable_attribute_name' => '[data-test-non-translatable-attribute-name="%name%"]',
+            'non_translatable_attribute_value' => '[data-test-non-translatable-attribute-value="%value%"]',
```
Comment (Wojdylak) [inline src/Sylius/Behat/Element/Product/ShowPage/AttributesElement.php]: I think that is unnecessary

File: src/Sylius/Bundle/AdminBundle/templates/product/show/content/page_body/details/configurable_product.html.twig [hunk 19]
```diff
@@ -14,15 +14,13 @@
             </thead>
             <tbody>
                 {% for variant in product.variants %}
-                    <tr>
+                    <tr {{ sylius_test_html_attribute('variants') }} {{ sylius_test_html_attribute('variant', "%s"|format(variant.code)) }}>
```
Comment (Wojdylak) [inline src/Sylius/Bundle/AdminBundle/templates/product/show/content/page_body/details/configurable_product.html.twig]:
```suggestion
<tbody {{ sylius_test_html_attribute('variants') }}>
                    <tr {{ sylius_test_html_attribute('variant', "%s"|format(variant.code)) }}>
```

File: src/Sylius/Bundle/AdminBundle/templates/product/show/content/page_body/details/shared/pricing/body/applied_promotions.html.twig [hunk 20]
```diff
@@ -1,9 +1,18 @@
...
-</td>
+{% if channel_pricing.appliedPromotions %}
+    <td class="text-end">
+        <div class="row">
+            {% for promotion in channel_pricing.appliedPromotions %}
+                <a class="mb-1"
+                    {{ sylius_test_html_attribute('applied-promotion') }}
+                    href="{{ path('sylius_admin_catalog_promotion_show', {'id': promotion.id}) }}"
+                    class="applied-promotion"
+                >
```
Comment (mpysiak) [inline src/Sylius/Bundle/AdminBundle/templates/product/show/content/page_body/details/shared/pricing/body/applied_promotions.html.twig]:
```suggestion
                <a class="mb-1 applied-promotion"
                    {{ sylius_test_html_attribute('applied-promotion') }}
                    href="{{ path('sylius_admin_catalog_promotion_show', {'id': promotion.id}) }}"
                >
```

File: features/admin/product/viewing_product_in_admin_panel/viewing_details_of_product_with_variants.feature [hunk 21]
```diff
@@ -17,13 +17,13 @@ Feature: Viewing details of a product with variants
...
 
-    @todo @ui @no-api
+    @no-api @ui
     Scenario: Viewing a configurable product
         When I access the "Iron Shield" product
         Then I should see product show page with variants
         And I should see product name "Iron Shield"
 
-    @todo @ui @api
+    @ui @api
```
Comment (Wojdylak) [inline features/admin/product/viewing_product_in_admin_panel/viewing_details_of_product_with_variants.feature]:
```suggestion
    @api @ui
```

**Review by Wojdylak — DISMISSED** [review]

**Review by Wojdylak — APPROVED** [review]

**Review by Rafikooo — APPROVED** [review]

**Pull request merged**